operator-common/src/main/java/io/strimzi/operator/common/model/Ca.java

```diff
@@ -610,13 +611,40 @@ public void createRenewOrReplace(String namespace, String clusterName, Map<Strin
             keyAnnotations.put(Annotations.ANNO_STRIMZI_IO_FORCE_REPLACE, Annotations.stringAnnotation(caKeySecret, Annotations.ANNO_STRIMZI_IO_FORCE_REPLACE, "false"));
         }
 
-        caCertSecret = secretCertProvider.createSecret(namespace, caCertSecretName, certData, Util.mergeLabelsOrAnnotations(labels, additionalLabels),
+        caCertSecret = createCaSecret(namespace, caCertSecretName, certData, Util.mergeLabelsOrAnnotations(labels, additionalLabels),
                 Util.mergeLabelsOrAnnotations(certAnnotations, additionalAnnotations), ownerRef);
 
-        caKeySecret = secretCertProvider.createSecret(namespace, caKeySecretName, keyData, labels,
+        caKeySecret = createCaSecret(namespace, caKeySecretName, keyData, labels,
                 keyAnnotations, ownerRef);
     }
 
+    /**
+     * Create a Kubernetes secret containing the provided secret data section
+     *
+     * @param namespace Namespace
+     * @param name Secret name
+     * @param data Map with secret data / files
+     * @param labels Labels to add to the Secret
+     * @param annotations annotations to add to the Secret
+     * @param ownerReference owner of the Secret
+     * @return the Secret
+     */
+    private static Secret createCaSecret(String namespace, String name, Map<String, String> data,
+                               Map<String, String> labels, Map<String, String> annotations, OwnerReference ownerReference) {
+        List<OwnerReference> or = ownerReference != null ? singletonList(ownerReference) : emptyList();
+        return new SecretBuilder()
+                .withNewMetadata()
+                .withName(name)
+                .withNamespace(namespace)
+                .withLabels(labels)
+                .withAnnotations(annotations)
+                .withOwnerReferences(or)
+                .endMetadata()
+                .withType("Opaque")
+                .withData(data)
+                .build();
+    }
```

Comment:
Maybe you can format the indentation to make it more readable?

```suggestion
        return new SecretBuilder()
                .withNewMetadata()
                    .withName(name)
                    .withNamespace(namespace)
                    .withLabels(labels)
                    .withAnnotations(annotations)
                    .withOwnerReferences(or)
                .endMetadata()
                .withType("Opaque")
                .withData(data)
                .build();
    }
```

Reply:
Done :)